src/containers/blocks.jsx

```diff
@@ -155,7 +155,8 @@ class Blocks extends React.Component {
                 (event.type === this.ScratchBlocks.Events.BLOCK_DELETE &&
                     event.oldJson.type === "procedures_definition") ||
                 (event.type === this.ScratchBlocks.Events.BLOCK_CREATE &&
-                    event.json.type === "procedures_definition")
+                    event.json.type === "procedures_definition" &&
+                    !event.recordUndo)
```

Comment:
This should probably have a comment cuz otherwise future peoples will have no idea why you're checking record undo.

Reply:
Good call, done!